File: MSS_Front/mss/src/app/api/models/respuesta-bean.ts
import { Codigo } from "./codigo";

export interface RespuestaBean {
    status: Codigo,
    mensaje: string,
    body?: string
}

export function getCodigo(respuesta: RespuestaBean): Codigo{
    switch (respuesta.status.toString()){
        case "OK":
            return Codigo.OK;
        case "CREADO":
            return Codigo.CREADO;
        case "ERROR":
            return Codigo.ERROR;
        case "NO_AUTORIZADO":
            return Codigo.NO_AUTORIZADO;
        case "NO_ENCONTRADO":
            return Codigo.NO_ENCONTRADO;
        default:
            return Codigo.ERROR;
    }
}

File: MSS_Back/mss/src/main/java/ar/edu/ubp/das/mss/models/RespuestaBean.java
package ar.edu.ubp.das.mss.models;

public class RespuestaBean {
    private Codigo status;
    private String mensaje;
    private String body;

    public RespuestaBean (){}

    public RespuestaBean (Codigo cod, String msj, String body){
        this.status = cod;
        this.mensaje = msj;
        this.body = body;
    }

    public Codigo getStatus() {
        return status;
    }
    public void setStatus(Codigo status) {
        this.status = status;
    }
    public String getMensaje() {
        return mensaje;
    }
    public void setMensaje(String mensaje) {
        this.mensaje = mensaje;
    }
    public String getBody() {
        return body;
    }
    public void setBody(String body) {
        this.body = body;
    }
}
